Modules/_testcapimodule.c

```diff
@@ -2587,6 +2587,117 @@ get_datetime_fromtimestamp(PyObject* self, PyObject *args)
     return rv;
 }
 
+static PyObject *
+test_PyDateTime_GET(PyObject *self, PyObject *args)
+{
+    PyObject *obj = NULL;
+    PyObject *rv;
+    int year, month, day;
+
+    if (!PyArg_ParseTuple(args, "O", &obj)) {
+        return NULL;
+    }
+
+    year = PyDateTime_GET_YEAR(obj);
+    month = PyDateTime_GET_MONTH(obj);
+    day = PyDateTime_GET_DAY(obj);
+
+    rv = PyTuple_New(3);
+    if (rv == NULL) {
+        return NULL;
+    }
+
+    PyTuple_SET_ITEM(rv, 0, PyLong_FromLong(year));
+    PyTuple_SET_ITEM(rv, 1, PyLong_FromLong(month));
+    PyTuple_SET_ITEM(rv, 2, PyLong_FromLong(day));
+
+    return rv;
```

Comment:
These 10 lines of code can be simplfiied using Py_BuildValue as a single line, Py_BuildValue handles errors for you.

```
return Py_BuildValue("iii", year, month, day);
```

(Same comment for the other added functions.)